### PR TITLE
Flag to enable workflows

### LINKED ISSUE #4761: Add feature flag to enable new workflows
Given `ServicesMain` in `hedera-app`, add a feature flag to enable using the new gRPC handlers and workflows, or the existing gRPC handlers. We can use this as a course-grained feature flag (since a fine-grained one appears to be very difficult to implement). It will simply switch from the existing gRPC handlers, to using the new handlers and workflows. Before this flag could be enabled in production, we would need to have all workflows complete, and we would need to have each transaction either implemented or delegating to the existing transaction handling code.

We may be able to add additional feature flags to enable each workflow independently. If so, that will be covered by other issues.

### PR DESCRIPTION
**Pull request by rbair23**
This PR adds a feature flag, "hedera.workflows.enabled" to enable the workflows pathway, from gRPC handling through the workflows and to the handlers. There is just one big flag that enables the pathway and applies to all workflows. With this PR, enabling the flag will take the new pathway, but that pathway *does not work*, and likely won't work for quite some time yet (it certainly won't pass tests).

The workflows need some interfaces which we need to find a way to get access to (maybe add them to `ServicesApp`). We will implement those in the workflow PRs.

With this PR, the `hedera-services/data/config/bootstrap.properties` now contains the flag, set to false. To make use of the new workflows, this needs to be set to true. But be sure to only set this flag locally in develop until we are ready to make it real (if you accidentally check it in with the flag set to true, it *will not* impact production, but it may impact our ability to do dev build/test). There is a similar flag for unit tests and for bdd tests.

Closes #4761 .

Signed-off-by: Richard Bair <rbair23@users.noreply.github.com>

### CODE REVIEW

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/ServicesMain.java [hunk 1]
```diff
@@ -116,7 +123,14 @@ private void exportAccountsIfDesired() {
     }
 
     private void startNettyIfAppropriate() {
-        app.grpcStarter().startIfAppropriate();
+        // The "hedera.workflows.enabled" feature flag indicates whether we enable the new gRPC
+        // server and workflows, or use the existing gRPC handlers in mono-service.
+        final var props = app.nodeLocalProperties();
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/ServicesMain.java:128]: Isn't this property a `GlobalDynamicProperty`?

File: hedera-node/hedera-mono-service/src/main/resources/bootstrap.properties [hunk 2]
```diff
@@ -207,3 +207,4 @@ hedera.prefetch.threadPoolSize=4
 hedera.prefetch.codeCacheTtlSecs=600
 utilPrng.isEnabled=true
 tokens.autoCreations.isEnabled=true
+hedera.workflows.enabled=false
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-mono-service/src/main/resources/bootstrap.properties:210]: We should probably add this to `hedera-mono-service/test/resource/standard.properties`, `test-clients/src/main/resource`.

**Review by Neeharika-Sompalli — APPROVED** [review]

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/ServicesApp.java [hunk 3]
```diff
@@ -133,6 +130,8 @@ public interface ServicesApp {
 
     GlobalDynamicProperties globalDynamicProperties();
 
+    GlobalStaticProperties globalStaticProperties();
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/ServicesApp.java:133]: Don't we need a provide method for this new class from Dagger to work for @Inject annotation in some module.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java [hunk 4]
```diff
@@ -31,20 +32,40 @@
...
 
-        // Setup and start the server
+        // Create the Ingest workflow. While we are in transition, some required facilities come
+        // from `hedera-app`, and some from `mono-service`. Eventually we'll transition all
+        // facilities to be from the app module.
+        final var ingestWorkflow =
+                new IngestWorkflowImpl(
+                        app.nodeInfo(),
+                        app.platformStatus(),
+                        null,
```
Comment (jsync-swirlds) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java:49]: All of these parameters require non-null.  This call will throw NPE.

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/ServicesApp.java [hunk 5]
```diff
@@ -23,10 +23,7 @@
 import com.hedera.node.app.service.mono.context.annotations.BootstrapProps;
 import com.hedera.node.app.service.mono.context.annotations.StaticAccountMemo;
 import com.hedera.node.app.service.mono.context.init.ServicesInitFlow;
-import com.hedera.node.app.service.mono.context.properties.GlobalDynamicProperties;
-import com.hedera.node.app.service.mono.context.properties.NodeLocalProperties;
-import com.hedera.node.app.service.mono.context.properties.PropertiesModule;
-import com.hedera.node.app.service.mono.context.properties.PropertySource;
+import com.hedera.node.app.service.mono.context.properties.*;
```
Comment (jsync-swirlds) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/ServicesApp.java:26]: Is this change to wildcard import needed/intended?

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/context/properties/GlobalStaticProperties.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+import static com.hedera.node.app.service.mono.context.properties.PropertyNames.WORKFLOWS_ENABLED;
+
+import com.hedera.node.app.service.mono.context.annotations.CompositeProps;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public class GlobalStaticProperties {
+    private final PropertySource properties;
```
Comment (tinker-michaelj) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/context/properties/GlobalStaticProperties.java:26]: For static properties, we only need to inject `BootstrapProperties` (which takes into account the JAR's _src/main/resources/bootstrap.properties_ plus any overrides from _data/config/bootstrap.properties_.)

This `@CompositeProps` instance is only needed to get any additional (dynamic) overrides from `0.0.121`.

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/context/properties/BootstrapProperties.java [hunk 7]
```diff
@@ -573,5 +574,6 @@ public static Function<String, Object> transformFor(String prop) {
                     entry(HEDERA_ALLOWANCES_IS_ENABLED, AS_BOOLEAN),
                     entry(ENTITIES_LIMIT_TOKEN_ASSOCIATIONS, AS_BOOLEAN),
                     entry(UTIL_PRNG_IS_ENABLED, AS_BOOLEAN),
-                    entry(TOKENS_AUTO_CREATIONS_ENABLED, AS_BOOLEAN));
+                    entry(TOKENS_AUTO_CREATIONS_ENABLED, AS_BOOLEAN),
+                    entry(WORKFLOWS_ENABLED, AS_BOOLEAN));
```
Comment (tinker-michaelj) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/context/properties/BootstrapProperties.java:578]: Imo this property would go more naturally in the `GLOBAL_STATIC_PROPS` section [here](https://github.com/hashgraph/hedera-services/blob/ea43d6109a15944bb92c7a870ae298f29bc017f8/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/context/properties/BootstrapProperties.java#L178) (tying into above comment).

**Review by tinker-michaelj — APPROVED** [review]
One suggestion to move `hedera.workflows.enabled` into the `GLOBAL_STATIC_PROPS` section and then source its value from `BootstrapProperties`.

But looks great!

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM! But, in the future if we want the `GlobalStaticProperties` to be injected to any other class, we will need a `Provides` annotated method.

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/context/properties/GlobalStaticProperties.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+package com.hedera.node.app.service.mono.context.properties;
+
+import static com.hedera.node.app.service.mono.context.properties.PropertyNames.WORKFLOWS_ENABLED;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public class GlobalStaticProperties {
```
Comment (hendrikebbers) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/context/properties/GlobalStaticProperties.java:25]: Why have you chosen `GlobalStaticProperties` as name? Do you plan to add more properties to this class? I would have assumed `FeatureFlagProperties` or something similar.

**Pull request merged**